### PR TITLE
[Mobile] - Fix missing custom gradient indicator in the color palette 

### LINKED ISSUE #42017: Selected custom gradient is not displayed in color settings
## Description
When setting a custom gradient in the color settings, there is no indication that a custom gradient is selected. The "Custom" indicator is missing entirely, leaving the appearance of no color/gradient selected. 

## Step-by-step reproduction instructions
1. Add a Buttons block. 
1. Open the settings for the Button block. 
1. Tap _Background_. 
1. Tap _Gradient_. 
1. Select a preset gradient. 
1. Tap _Customize Gradient_. 
1. Modify the gradient. 
1. Tap _Back_. 

## Expected behaviour
At the end of the list of gradients, a "Custom" option is listed and marked as selected with a checkmark. 

## Actual behaviour
There is no "Custom" option listed in the list of gradients. 

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/438664/176231648-36002fcc-348b-44fc-809c-0e29ebe668aa.MP4

## WordPress information
- WordPress version: 6.0
- Gutenberg version: 13.5.0
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Geologist

## Device information
- Device: iPhone SE, Samsung Galaxy S20
- Operating system: iOS 15.6, Android 12
- WordPress app version: 20.1

### PR DESCRIPTION
**Pull request by geriux**
Fixes https://github.com/WordPress/gutenberg/issues/42017

## What?
This PR fixes a regression most likely introduced in https://github.com/WordPress/gutenberg/pull/38417 where we added support for multiple color palettes where the custom color indicator is not shown for custom gradient colors.

## Why?
To fix a UI regression and possibly avoid confusion for users when they customize their gradient colors.

## How?
The custom indicator wasn't being shown due to not having all of the gradient colors available for the current color segment being rendered.

We also limit the custom indicator to only be visible in the first available color palette row, since users can have up to three: `Default`, `Theme`, and `Custom` colors.

## Testing Instructions

> [!NOTE]  
> These changes need to be tested using the main host apps and a theme that has custom gradients e.g: [Twenty Twenty-Four](https://wordpress.org/themes/twentytwentyfour/)

- Add a Buttons block.
- Open the settings for the Button block.
- Tap Background.
- Tap Gradient.
- Select a preset gradient.
- Tap Customize Gradient.
- Modify the gradient.
- Tap Back
- Expect to see the Custom indicator at the end of the list
- Tap Back again
- Tap Background
- Expect to see the list to auto-scroll to the end where the Custom indicator should be visible

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
https://github.com/WordPress/gutenberg/assets/4885740/66d90cdb-f11c-4590-a932-0d266de36683

### CODE REVIEW

File: packages/block-library/src/buttons/test/edit.native.js [hunk 1]
```diff
@@ -391,5 +391,59 @@ describe( 'Buttons block', () => {
...
+			fireEvent.press( buttonBlock );
+
+			// Open Block Settings.
+			fireEvent.press( screen.getByLabelText( 'Open Settings' ) );
+
+			// Wait for Block Settings to be visible.
+			const blockSettingsModal = screen.getByTestId(
+				'block-settings-modal'
+			);
+			await waitFor( () => blockSettingsModal.props.isVisible );
```
Comment (fluiddot) [inline packages/block-library/src/buttons/test/edit.native.js]: Nitpick: We could use the test helper [`openBlockSettings`](https://github.com/WordPress/gutenberg/blob/trunk/test/native/integration-test-helpers/open-block-settings.js).

File: packages/components/src/color-palette/index.native.js [hunk 2]
```diff
@@ -110,7 +121,7 @@ function ColorPalette( {
 
 	function isSelectedCustom() {
 		const isWithinColors =
-			activeColor && mergedColors && mergedColors.includes( activeColor );
+			activeColor && allColors && allColors.includes( activeColor );
```
Comment (fluiddot) [inline packages/components/src/color-palette/index.native.js]: Nitpick: we could simplify it by removing one of the conditions.
```suggestion
			activeColor && allColors?.includes( activeColor );
```

File: packages/components/src/mobile/color-settings/palette.screen.native.js [hunk 3]
```diff
@@ -57,6 +55,10 @@ const PaletteScreen = () => {
 	const currentSegmentColors = ! isGradientSegment
 		? defaultSettings.colors
 		: defaultSettings.gradients;
+	const allAvailableColors = useMobileGlobalStylesColors();
+	const allAvailableGradients = currentSegmentColors
+		.map( ( item ) => item.gradients || [] )
+		.reduce( ( acc, val ) => acc.concat( val ), [] );
```
Comment (fluiddot) [inline packages/components/src/mobile/color-settings/palette.screen.native.js]: Nitpick: we could simplify this by using `flatMap`.
```suggestion
	const allAvailableGradients = currentSegmentColors
		.flatMap( ( { gradients } ) => gradients )
		.filter( Boolean );
```

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! 

I added some suggestions but none shouldn't block the merge of the PR.

File: packages/block-library/src/buttons/test/edit.native.js (hunk 1 above)
Comment (geriux) [reply packages/block-library/src/buttons/test/edit.native.js]: Thanks for the suggestion! Updated it in https://github.com/WordPress/gutenberg/pull/57605/commits/3fdabed0594fd75fcb5ac15135ab00227bbcd82e

File: packages/components/src/color-palette/index.native.js (hunk 2 above)
Comment (geriux) [reply packages/components/src/color-palette/index.native.js]: Nice! https://github.com/WordPress/gutenberg/pull/57605/commits/dd7b91a4df40180b6bb9d420bb94081679bef46e

File: packages/components/src/mobile/color-settings/palette.screen.native.js (hunk 3 above)
Comment (geriux) [reply packages/components/src/mobile/color-settings/palette.screen.native.js]: Updated in https://github.com/WordPress/gutenberg/pull/57605/commits/32466822ae1da7b17fad76139832ea8f8dfb3354

**Pull request merged**